### PR TITLE
Support  MySQL `ROWS` syntax for `VALUES`

### PR DESCRIPTION
**Pull request by aljazerzen**
MySQL does not support following query:

```sql
VALUES (1)
```
... but supports ...

```sql
VALUES ROW(1)
```
https://dev.mysql.com/doc/refman/8.0/en/values.html

Note that is does support:
```sql
INSERT INTO my_table VALUES (1)
```

Quality of my code reflects banality of this detail.

### CODE REVIEW

File: src/ast/query.rs [hunk 1]
```diff
@@ -709,16 +709,20 @@ impl fmt::Display for Top {
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Values(pub Vec<Vec<Expr>>);
+pub struct Values {
+    pub explicit_row: bool,
```
Comment (alamb) [inline src/ast/query.rs:713]:
```suggestion
    /// Was there an explict ROWs keyword (MySQL)?
    /// <https://dev.mysql.com/doc/refman/8.0/en/values.html>
    pub explicit_row: bool,
```

File: src/parser.rs [hunk 2]
```diff
@@ -5482,14 +5490,25 @@ impl<'a> Parser<'a> {
...
-    pub fn parse_values(&mut self) -> Result<Values, ParserError> {
-        let values = self.parse_comma_separated(|parser| {
+    pub fn parse_values(&mut self, within_insert: bool) -> Result<Values, ParserError> {
+        let mut explicit_row = false;
+
+        let rows = self.parse_comma_separated(|parser| {
+            if parser.parse_keyword(Keyword::ROW) {
+                explicit_row = true;
+            } else {
+                if !within_insert && parser.dialect.values_require_row_in_select() {
```
Comment (alamb) [inline src/parser.rs:5500]: How important is it to reject the ROWS keyword during inserts?  I see that mySQL is not consistent in that ROW is not allowed in INSERT statements https://dev.mysql.com/doc/refman/8.0/en/insert.html

However, I also don't see any (negative) tests for this case in this PR, though maybe I have missed them

My preference is to remove the special case handling for in select and make the parser permissive and support an optional ROW whenever a VALUES list is specified.

Downstream crates that care can detect `explicit_row` and error if they don't want to allow that syntax.

We tried to make the philosophy of "parse all the things" clearer recently in the introduction here https://github.com/sqlparser-rs/sqlparser-rs#extensible-sql-lexer-and-parser-for-rust

**Review by alamb — COMMENTED** [review]
Thank you @aljazerzen 

The code and tests look good except for the `within_insert` special case

File: src/parser.rs (hunk 2 above)
Comment (alamb) [reply src/parser.rs:5500]: BTW I found

```
Quality of my code reflects banality of this detail.
```

😆

Comment (aljazerzen) [reply src/parser.rs:5500]: Yeah that makes sense - there is no need to inforce explict row. And it simplifies the code a lot!

**Review by alamb — APPROVED** [review]
Looks good -- thank you @aljazerzen

**Pull request merged**